crates/gen/src/types/delegate.rs

```diff
@@ -58,6 +58,32 @@ impl Delegate {
             .map(|param| param.to_invoke_arg_tokens());
         let debug = debug::default_debug_tokens(&self.name);
 
+        let invoke_upcall = if let Some(return_type) = &self.method.return_type {
```

Comment:
This is just roughed in to allow the resulting Rust code to compile - it doesn't actually work yet. That's up next and I'm sure will change.